skbio/io/format/stockholm.py

```diff
@@ -577,3 +576,79 @@ def _check_for_malformed_line(line, expected_len):
         raise StockholmFormatError('Line contains %d item(s). It must '
                                    'contain exactly %d item(s).'
                                    % (len(line), expected_len))
+
+
+@stockholm.writer(TabularMSA)
+def _tabular_msa_to_stockholm(obj, fh):
+    index = list(obj.index)
+    # Writes header
+    fh.write("# STOCKHOLM 1.0\n")
+
+    # Writes GF data to file
+    if obj.has_metadata():
+        for i in range(len(obj.metadata)):
+            gf_feature = list(obj.metadata.keys())[i].strip("'")
+            if gf_feature == 'NH' and (type(obj.metadata[gf_feature]) is
+                                       OrderedDict):
+                tree_data = obj.metadata[gf_feature]
+                for j in range(len(tree_data)):
+                    gf_feature_data = list(tree_data.keys())[j]
+                    fh.write("#=GF TN %s\n" % gf_feature_data)
+                    gf_feature_data = list(tree_data.values())[j]
+                    fh.write("#=GF NH %s\n" % gf_feature_data)
+            else:
+                gf_feature_data = obj.metadata[gf_feature].strip("'")
+                fh.write("#=GF %s %s\n" % (gf_feature, gf_feature_data))
+
+    # Writes GS data to file
+    for i in range(len(obj)):
+        metadata = obj[i].metadata
```

Comment:
Handle missing metadata efficiently (one of the unit tests I asked you to add will catch this bug):

```python
if seq.has_metadata():
    ...
```